instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex

```diff
@@ -240,4 +242,14 @@ defmodule OpentelemetryEcto do
   defp add_additional_attributes(attributes, additional_attributes) do
     Map.merge(attributes, additional_attributes)
   end
+
+  defp telemetry_prefix(event_prefix) when is_list(event_prefix) do
+    event_prefix
+  end
+
+  defp telemetry_prefix(repo) when is_atom(repo) do
+    repo
+    |> Kernel.apply(:config, [])
```

Comment:
If you call `repo.config([])` here instead it'll already make it easier for the Elixir type system to detect if `repo` is a module and so on.

Reply:
It's one of those things that you do on Friday, and when you come back on Monday, you ask, "What was I thinking here?"

Thank you 💜